### PR TITLE
Video.js 6 Compatibility

### PR DESCRIPTION
**Pull request by misteroneill**
This adds compatibility for Video.js 6 while maintaining backward-compatibility with Video.js 5.

### CODE REVIEW

File: test/test.macros.js [hunk 1]
```diff
@@ -47,14 +47,18 @@ QUnit.test('timestamp', function(assert) {
 QUnit.test('document.referrer', function(assert) {
   var result = this.player.ads.adMacroReplacement('{document.referrer}');
 
-  assert.ok(result.match(/http:\/\/localhost:\d+\/\?id=\d+/),
+  assert.equal(
+    result,
+    document.referrer,
```
Comment (misteroneill) [inline test/test.macros.js]: This was necessary when running tests via a browser tab (rather than `npm test`).

File: test/test.macros.js [hunk 2]
```diff
@@ -47,14 +47,18 @@ QUnit.test('timestamp', function(assert) {
...
     '"' + result + '" was the document.referrer');
 });
 
 QUnit.test('window.location.href', function(assert) {
   var result = this.player.ads.adMacroReplacement('{window.location.href}');
 
-  assert.ok(result.match(/http:\/\/localhost:\d+\/context.html/),
+  assert.equal(
+    result,
+    window.location.href,
```
Comment (misteroneill) [inline test/test.macros.js]: This was necessary when running tests via a browser tab (rather than `npm test`).

File: test/test.snapshot.js [hunk 3]
```diff
@@ -78,7 +83,12 @@ QUnit.test('tries to restore the play state up to 20 times', function(assert) {
 
   // We expect 20 timeouts at 50ms each.
   this.clock.tick(1000);
-  assert.strictEqual(setTimeoutSpy.callCount, 20, 'seekable was tried multiple times');
+
+  // More aspects of the Video.js 6 player are asynchronous. This accounts for
+  // that difference.
+  var expectedCount = (videojs.use) ? 23 : 20;
```
Comment (incompl) [inline test/test.snapshot.js]: This test is not effective. I'd rather remove it completely.

File: src/plugin.js [hunk 4]
```diff
@@ -633,5 +633,7 @@ const contribAdsPlugin = function(options) {
 
 };
 
+const registerPlugin = videojs.registerPlugin || videojs.plugin;
```
Comment (gkatsev) [inline src/plugin.js]: much more succinct than what I used in contrib-hls. 👍

File: test/test.snapshot.js (hunk 3 above)
Comment (misteroneill) [inline test/test.snapshot.js]: Done.

File: test/test.macros.js (hunk 2 above)
Comment (incompl) [inline test/test.macros.js]: Looks good to me!

**Comment by incompl** [comment]
LGTM

**Pull request merged**